dotCMS/src/main/java/com/dotcms/util/pagination/ThemePaginator.java

```diff
@@ -1,75 +1,124 @@
-        final String query = String.format(hostId == null ? LUCENE_QUERY_WITH_OUT_HOST : LUCENE_QUERY_WITH_HOST, hostId);
+        final String hostId       = params != null ? (String) params.get(HOST_ID_PARAMETER_NAME) : null;
+        final String searchParams = params != null ? (String) params.get(SEARCH_PARAMETER) : null;
+        final String searchById   = params != null ? (String) params.get(IDENTIFIER_PARAMETER) : null;
+        StringBuilder query       = new StringBuilder();
+
+        if (UtilMethods.isSet(searchById)){
+            query.append(BASE_LUCENE_QUERY);
+            query.append(" +");
+            query.append(IDENTIFIER_PARAMETER);
+            query.append(StringPool.COLON);
+            query.append(searchById);
+        }else{
+            query.append(String.format(
+                    hostId == null ? LUCENE_QUERY_WITHOUT_HOST : LUCENE_QUERY_WITH_HOST, hostId));
+        }
+
+        if (UtilMethods.isSet(searchParams)){
+            query.append(" +catchall:");
+            query.append(searchParams);
+            query.append("*");
+        }
+
         final String sortBy = String.format("parentPath %s", direction.toString().toLowerCase());
 
         try {
 
-            final PaginatedArrayList<ContentletSearch> contentletSearches = (PaginatedArrayList<ContentletSearch>)
-                    contentletAPI.searchIndex(query, limit, offset, sortBy, user, false);
+            final List<ContentletSearch> contentletSearches = (PaginatedArrayList<ContentletSearch>)
+                    contentletAPI.searchIndex(query.toString(), limit, offset, sortBy, user, false);
 
             final List<String>  inodes = contentletSearches.stream()
                     .map(contentletSearch -> contentletSearch.getInode())
                     .collect(Collectors.toList());
 
-            final PaginatedArrayList<Contentlet> result = new PaginatedArrayList();
-            result.setTotalResults(contentletSearches.getTotalResults());
-            result.setQuery(query);
-            result.addAll(contentletAPI.findContentlets(inodes));
+            final PaginatedArrayList<Folder> result = new PaginatedArrayList();
+
+            for (Contentlet contentlet :contentletAPI.findContentlets(inodes)) {
+                Folder folder = folderAPI.find(contentlet.getFolder(), user, false);
+                folder.setIdentifier(getThemeIdentifier(folder, user));
+                result.add(folder);
+            }
+
+            result.setTotalResults(result.size());
+            result.setQuery(query.toString());
             return result;
         } catch (DotSecurityException | DotDataException e) {
             throw new PaginationException(e);
         }
     }
+
+    @VisibleForTesting
+    String getThemeIdentifier(Folder folder, User user) throws DotSecurityException, DotDataException {
+
+        StringBuilder query = new StringBuilder();
```

Comment:
It would be great to have a QueryBuilder such as

query.addTitle (THEME_PNG);
or addProperty("conFolder", folder.getInode);

could be cleaner and helpful to reuse common ES fields